### DISCUSSION TITLE
`LazyGhostTrait` on entity with magic `__get()`

### DISCUSSION THREAD

**Discussion by jahudka** [symfony/symfony#49434] [Q&A] [answered]
I have a trait which i routinely apply to all my entities which provides a magic `__get()` implementation which forwards access to entity properties to getters, if they exist - e.g. `echo $entity->someProp;` would be forwarded to `$entity->getSomeProp()`, if that exists, and throw an exception if it doesn't.

I have upgraded one of my projects to Symfony 6.2 and this broke down hard when accessing properties of lazy-loaded associations, e.g. `echo $entity->relatedEntity->someProp;` - it throws an `Typed property RelatedEntity::$someProp must not be accessed before initialization` exception. As I understand, this is because the `LazyGhostTrait` relies on _property access_ to figure out when to load a lazy object, and since I'm using `__get()`, there is no property access happening until inside the getter which `__get()` calls.. so the related entity isn't loaded and things crash and burn.

Did anyone ever run into something like this? Does it have a solution at all? I'm assuming the `LazyGhostTrait` implementation also uses magic accessors, so it just might not be possible to do..

**Comment by stof** [comment]
If you want to access your entity properties publicly, the best solution is probably to make them public for real. The proxies based on LazyGhostTrait will work fine with public properties.

**Reply by jahudka** [reply]
Well, I think that's not really a solution to my problem, since one other thing the trait does is translate `$entity->property = 'value';` to `$entity->setProperty('value');` and having _smart_ setters is something I don't want to lose. But whether my use-case is legitimate or not, I haven't found even a single mention of the Symfony traits possibly not working when other magic methods are involved, and I think there should probably be a loud warning about that somewhere.

**Reply by griesi007** [reply]
@jahudka Did you find a solution. I am fighting with the same problem.

**Reply by jahudka** [reply, marked as answer]
I'm honestly not sure anymore; I probably gave up. Might be a different case these days with PHP 8.4 property hooks, but I haven't looked into that yet.
